server/gitrest/src/utils/isomorphicgitManager.ts

```diff
@@ -0,0 +1,394 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as isomorphicGit from "isomorphic-git";
+import winston from "winston";
+import safeStringify from "json-stringify-safe";
+import type * as resources from "@fluidframework/gitresources";
+import { NetworkError } from "@fluidframework/server-services-client";
+import * as helpers from "./helpers";
+import * as conversions from "./isomorphicgitConversions";
+import {
+    IRepositoryManagerFactory,
+    IExternalWriterConfig,
+    IRepositoryManager,
+    IFileSystemManager,
+} from "./definitions";
+
+export class IsomorphicGitRepositoryManager implements IRepositoryManager {
+    constructor(
+        private readonly fileSystemManager: IFileSystemManager,
+        private readonly repoOwner: string,
+        private readonly repoName: string,
+        private readonly directory: string,
+    ) {}
+
+    public get path(): string {
+        return this.directory;
+    }
+
+    public async getCommit(sha: string): Promise<resources.ICommit> {
+        const commit = await isomorphicGit.readCommit({
+                fs: this.fileSystemManager,
+                gitdir: this.directory,
+                oid: sha,
+            });
+        return conversions.commitToICommit(commit);
+    }
+
+    public async getCommits(
+        sha: string,
+        count: number,
+        externalWriterConfig?: IExternalWriterConfig,
+    ): Promise<resources.ICommitDetails[]> {
+        try {
+            const commits = await isomorphicGit.log({
+                    fs: this.fileSystemManager,
+                    gitdir: this.directory,
+                    ref: sha,
+                    depth: count,
+                });
+
+             return commits.map((rawCommit) => {
+                const gitCommit = conversions.commitToICommit(rawCommit);
+                const result: resources.ICommitDetails =
+                {
+                    commit: {
+                        author: gitCommit.author,
+                        committer: gitCommit.committer,
+                        message: gitCommit.message,
+                        tree: gitCommit.tree,
+                        url: gitCommit.url,
+                    },
+                    parents: gitCommit.parents,
+                    sha: gitCommit.sha,
+                    url: "",
+
+                };
+                return result;
+            });
+        } catch (err) {
+            winston.info(`getCommits error: ${err}`);
+            return Promise.reject(err);
```

Comment:
throw 500?

Reply:
Good point. `NodeGitManager` has the same issue (rejecting promise instead of throwing `NetworkError`), but the logic there is a bit more complicated since it has many if/else case scenarios due to `externalWriterConfig`/`externalStorageManager`, so I will leave it as-is until we revisit `NodeGitManager` to possibly remove support for external storage (or until we just go there to improve it 😄 )